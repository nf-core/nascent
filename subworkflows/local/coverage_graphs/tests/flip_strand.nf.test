nextflow_workflow {

    name "Test COVERAGE_GRAPHS"
    script "../main.nf"
    workflow "COVERAGE_GRAPHS"

    // Required test data
    test("Should flip strands for PROseq assay") {

        when {
            params {
                assay_type = "PROseq"
            }

            workflow {
                """
                input[0] = Channel.of([
                    [ id:'test_sample', strandedness:'forward' ],
                    file("https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam"),
                    file("https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam.bai")
                ])
                input[1] = file("https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/modules/data/genomics/homo_sapiens/genome/chr21/sequence/genome.sizes")
                input[2] = file("https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/modules/data/genomics/homo_sapiens/genome/chr21/sequence/genome.fasta")
                input[3] = file("https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/modules/data/genomics/homo_sapiens/genome/chr21/sequence/genome.fasta.fai")
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.trace).match("flipped_trace") },
                { assert snapshot(workflow.out.plus_minus.get(0)[1]).match("flipped_bigwig") },
                { assert snapshot(path(workflow.out.versions.get(0)).yaml).match("flipped_versions") }
            )
        }
    }

    test("Should NOT flip strands for GROseq assay") {

        when {
            params {
                assay_type = "GROseq"
            }

            workflow {
                """
                input[0] = Channel.of([
                    [ id:'test_sample', strandedness:'forward' ],
                    file("https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam"),
                    file("https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam.bai")
                ])
                input[1] = file("https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/modules/data/genomics/homo_sapiens/genome/chr21/sequence/genome.sizes")
                input[2] = file("https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/modules/data/genomics/homo_sapiens/genome/chr21/sequence/genome.fasta")
                input[3] = file("https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/modules/data/genomics/homo_sapiens/genome/chr21/sequence/genome.fasta.fai")
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.trace).match("nonflipped_trace") },
                { assert snapshot(workflow.out.plus_minus.get(0)[1]).match("nonflipped_bigwig") },
                { assert snapshot(path(workflow.out.versions.get(0)).yaml).match("nonflipped_versions") }
            )
        }
    }
}
