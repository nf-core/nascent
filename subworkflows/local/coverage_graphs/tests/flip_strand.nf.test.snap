{
    "nonflipped_bigwig": {
        "content": [
            "test_sample.plus.bigWig:md5,f9497639d1179b9145d2b0d9572c6650"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-05-04T11:45:03.268586"
    },
    "nonflipped_versions": {
        "content": [
            {
                "COVERAGE_GRAPHS:DEEPTOOLS_BAMCOVERAGE_MINUS": {
                    "deeptools": "3.5.5"
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-05-04T11:45:03.273502"
    },
    "flipped_trace": {
        "content": [
            {
                "tasksFailed": 0,
                "tasksCount": 3,
                "tasksSucceeded": 3
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-05-04T11:44:56.781572"
    },
    "nonflipped_trace": {
        "content": [
            {
                "tasksFailed": 0,
                "tasksCount": 2,
                "tasksSucceeded": 2
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-05-04T11:45:03.265038"
    },
    "flipped_bigwig": {
        "content": [
            "test_sample.flipped.bigWig:md5,a93df4c511c8344407434f718837dd64"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-05-04T11:44:56.792779"
    },
    "flipped_versions": {
        "content": [
            {
                "COVERAGE_GRAPHS:DEEPTOOLS_BAMCOVERAGE_MINUS": {
                    "deeptools": "3.5.5"
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-05-04T11:44:56.872212"
    }
}